sql/moz-fx-data-shared-prod/telemetry_derived/cohort_daily_stats_clients_frst_seen_v2/query.sql

```diff
@@ -0,0 +1,167 @@
+    attribution_content,
+    attribution_experiment,
+    attribution_medium,
+    attribution_source,
+    attribution_ua,
+    -- attribution_variation, -- can find in clients_daily_v6 with unpacking attribution field
+    partner_distribution_version,
+    partner_distributor,
+    partner_distributor_channel,
+    partner_id,
+    city,
+    country,
+    -- device_model, -- important for mobile, usually null, may not need
+    apple_model_id, -- from clients_first_seen_v2
+    distribution_id,
+    -- is_default_browser,  -- can find in clients_daily_v6
+    locale,
+    -- normalized_app_name,  -- strings that come from unified_metrics/filters out BrowserStack
+    app_name as normalized_app_name, -- from clients_first_seen_v2, as per Lucia's msg app_name = normalized_app_name
+    normalized_channel,
+    normalized_os,
+    normalized_os_version,
+    -- os_version_major,
+    -- os_version_minor,
+    COALESCE(
+      SAFE_CAST(NULLIF(SPLIT(normalized_os_version, ".")[SAFE_OFFSET(0)], "") AS INTEGER),
+      0
+    ) AS os_version_major,
+    COALESCE(
+      SAFE_CAST(NULLIF(SPLIT(normalized_os_version, ".")[SAFE_OFFSET(1)], "") AS INTEGER),
+      0
+    ) AS os_version_minor,
+  FROM
+    -- telemetry_derived.rolling_cohorts_v1
+    telemetry_derived.clients_first_seen_v2
+  WHERE
+    -- cohort_date
+    first_seen_date
+    BETWEEN DATE_SUB(DATE("2023-10-01"), INTERVAL 180 DAY)
+    AND DATE_SUB(DATE("2023-10-01"), INTERVAL 1 DAY)
+    -- (1) No need to get activity for the cohort created on activity_date - everyone will be retained
+    -- (2) Note this is a pretty big scan... Look here for problems
+),
+activity_cohort_match AS (
+  SELECT
+    cohorts_in_range.client_id AS cohort_client_id,
+    submission_date_activity.client_id AS active_client_id,
+    submission_date_activity.days_seen_bits as active_client_days_seen_bits,
+    submission_date_activity.days_visited_1_uri_bits,
+    submission_date_activity.days_interacted_bits,
+    mozfun.bits28.days_since_seen(submission_date_activity.days_seen_bits) as active_clients_days_since_seen,
+    cohorts_in_range.* EXCEPT (client_id)
+  FROM
+    cohorts_in_range
+  LEFT JOIN
+    submission_date_activity
+  USING
+    (client_id, activity_date)
+)
+SELECT
+  cohort_client_id,
+  sample_id,
+  cohort_date,
+  second_seen_date,
+  activity_date,
+  -- activity_segment,
+  app_version,
+  attribution_campaign,
+  attribution_content,
+  attribution_experiment,
+  attribution_medium,
+  attribution_source,
+  attribution_ua,
+  -- attribution_variation,
+  partner_distribution_version,
+  partner_distributor,
+  partner_distributor_channel,
+  partner_id,
+  city,
+  country,
+  -- device_model,
+  apple_model_id, -- from clients_first_seen_v2
+  distribution_id,
+  -- is_default_browser,
+  locale,
+  normalized_app_name,
+  normalized_channel,
+  normalized_os,
+  normalized_os_version,
+  os_version_major,
+  os_version_minor,
+  COUNT(cohort_client_id) AS num_clients_in_cohort,
```

Comment:
What's the rationale for materializing this column? Wouldn't not having it and doing a count(*) or count(client_id) at analysis work the same?

Reply:
This is supposed to be an aggregated table at the end. Should not have selected client_id and sample_id as columns for the table. My mistake